perf/perf_c2c.py

```diff
@@ -0,0 +1,119 @@
+    the help of yaml file
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+
+        if 'ppc64' not in detected_distro.arch:
+            self.cancel('This test is not supported on %s architecture'
+                        % detected_distro.arch)
+
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in self.distro_name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s' %
+                         platform.uname()[2]])
+        elif self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported \
+                         by %s" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        # Check for c2c is available in the system.
+        output = process.run('perf mem record -e list')
+        if 'ldlat-stores' or 'ldlat-loads' in output:
+            self.log.info("perf c2c is available")
+        else:
+            self.cancel('perf c2c is not available')
+
+        # Creating temporary file to collect the perf.data
+        self.temp_file = tempfile.NamedTemporaryFile().name
+
+        # Getting the parameters from yaml file
+        self.record = self.params.get('record_method', default='record')
+        self.report = self.params.get('report_method', default='report')
+
+        # Clear the dmesg, by that we can capture the delta at the end of the test.
+        process.run("dmesg -C", sudo=True)
+
+    def verify_dmesg(self):
+        self.whiteboard = process.system_output("dmesg")
+        pattern = ['WARNING: CPU:', 'Oops',
+                   'Segfault', 'soft lockup', 'Unable to handle']
+        for fail_pattern in pattern:
+            if fail_pattern in self.whiteboard:
+                self.fail("Test Failed : %s in dmesg" % fail_pattern)
+
+    def run_cmd(self, cmd):
+        try:
+            process.run(cmd, ignore_status=False, sudo=True)
+        except process.CmdError as details:
+            self.fail("Command %s failed: %s" % (cmd, details))
+
+    def test_c2c(self):
+        # When input is used for report, then need to pass the file argument
+        # to the same file, record should log the data. So altering record,
+        # report options to have the proper arguments.
+        if self.report == "-i":
+            self.report = "-i %s" % self.temp_file
+            self.record = self.record + " -o %s" % self.temp_file
+        if self.report == "--input":
+            self.report = "--input=%s" % self.temp_file
+            self.record = self.record + " -o %s" % self.temp_file
```

Comment:
self.record is assigned same in both the cases. Can they be outside the if block ?

Reply:
@narasimhan-v `self.record` should be assigned with `-o` option only in the case of `report's -i or --input` option is present so has to be inside the `if condition`.

Reply:
ok, got it